### ISSUE TITLE
linter: no-invalid-fetch-options

### ISSUE THREAD

**Issue by CapySloth** [oxc-project/oxc#9405]
### What version of Oxlint are you using?

0.15.12

### What happened?

Seeing no-invalid-fetch-options trigger when using an enum that maps to "POST", which does not seem like intended behaviour.

```tsx
export declare enum Method {
    Post = "POST",
}
 const response = await fetch("/", {
    method: Method.Post,
    body: "",
  });
```

The linter interprets enum as a "GET" method, which is incorrect.
`eslint-plugin-unicorn(no-invalid-fetch-options): "body" is not allowed when method is "GET"`

There is a chance this is intentional and the rule intends to push folks that opt-into it to only use hardcoded strings for fetch methods.

**Comment by therewillbecode** [comment]
Fix implemented here:
https://github.com/oxc-project/oxc/pull/9416
